meson.build

```diff
@@ -338,6 +338,21 @@ data_files = files(
 	'data/devices/steelseries-sensei-raw.device',
 )
 
+illegal_filename_characters = ['[', ']']
+# Validate installed filenames for resources to ensure we don't build/ship with
+# characters that downstream package management tools aren't happy with.
+foreach data_file : data_files
+	data_file_str = '@0@'.format(data_file)
+	foreach illegal_character : illegal_filename_characters
+		if data_file_str.contains(illegal_character)
+			error('illegal character "@1@" found in data file "@0@"'.format(
+				data_file_str, illegal_character
+			))
+			break
+		endif
+	endforeach
+endforeach
```

Comment:
This should be in a unit test, not the meson file.

Can you move it to `data/devices/data-parse-test.py`?

Reply:
Easy enough. Any complaints if I use a regular expression to do that test?

Reply:
It should be fine :grin:.

Reply:
I've moved it out of the build file, and squashed the commits.